Online Monitoring tool utilising corryvreckan

s6anloes opened the pull request:
Addition of a new tool which allows monitoring of test beam data recorded using EUDAQ2 as DAQ control system to be monitored (semi-)online by starting a corryvreckan session and utilising its OnlineMonitor module. Since this project also required changes in corryvreckan, it only works with my recent changes to corry for which I am creating a merge request in parallel.

The changes to the repository are mainly the addition of new files:

1. `user/example/misc/CorryMon.conf`: 
The main source file in which everything is sorted to start the corry session correctly
2. Various files which run the monitoring with the EUDAQ2 example Producer & DataCollector
   - `user/example/misc/CorryMon.ini`: EUDAQ2 ini file for running the example
   - `user/example/misc/CorryMon.conf`: EUDAQ2 conf file for running the example
   - `user/example/misc/corryconfig.conf`: corryvreckan configuration file to run the example
   - `user/example/misc/corrygeo.geo`: corryvreckan geometry file for Ex0Producer
   - `user/example/misc/corrymon_startrun.sh`: script to start the monitoring example
3. Two changes (additions) to the core library source code
   - `main/lib/core/include/eudaq/Configuration.hh`: Added function to get sections of a configuration (analog to `Keylist()`)
   - `main/lib/core/src/Utils.cc`: Adding error to being unable to open a file for reading

lhuth commented:
@herkerta Maybe you also want to crosscheck

simonspa commented on main/lib/core/src/Utils.cc:
```diff
@@ -155,7 +155,7 @@ namespace eudaq {
         EUDAQ_THROW("Error reading from file " + fname);
       }
-    }
+    } else EUDAQ_THROW("Unable to open file " + fname + " for reading");
```
Suggest adding brackets.

s6anloes commented:
- Added brackets to Utils.cc
- Went back to calling corry directly due to thread safety converns
- Added README for running the CorryMonitor examples

arummler commented:
Why is this supposed to live under examples and not under monitors? I did not have the impression that it has to be adapted to a specific set of DAQs included in the run?

s6anloes commented:
1. In the beginning I was just aligning my code with the Ex0Monitor for structure. There is no real reason why it should be in examples other than that it will be compiled by default in this case. But if there is no strong objection (@lhuth) I can move it to a dedicated folder in the monitors directory.
2. Sorry, can you point out which CMakeLists.txt is missing?
3. Were you thinking something along the lines of [this](https://solarianprogrammer.com/2019/01/13/cpp-17-filesystem-write-file-watcher-monitor/)? I can certainly try to implement this

s6anloes commented:
Closing this pull request and opening a new one with the requested changes.
I had some git problems and couldn't figure out how to include all recent changes in the main repo into this branch.
A new request should fix this problem
